pkg/upgrade/engine/v1alpha1/upgrade_template_engine.go

```diff
@@ -0,0 +1,131 @@
+/*
```

Comment:
@imazik 
IMO we should try to avoid creating a new castemplate engine.
A lot of engines were created as the core engine was lacking features.
Let us try to add features to core engine or refactor the core engine into a new version package.
The new core engine can be used directly from upgrade executor etc.
The main binary of task executor (i.e. main.go) can possibly directly execute this new core engine.